versions/next/asyncapi.md

```diff
@@ -2208,6 +2231,68 @@ petstore_auth:
 - read:pets
 ```
 
+### <a name="correlationIdObject"></a>Correlation ID Object
+
+An object that specifies an identifier at design time that can used for message tracing and correlation.
+
+For specifying and computing the location of a Correlation ID a [runtime expression](#runtimeExpression) is used.
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---|---
+description | `string` | A optional description of the identifier. [CommonMark syntax](http://spec.commonmark.org/) can be used for rich text representation.
+type | `integer` \| `string` \| `number` | An optional type specification.
+location | {expression} | **REQUIRED.** A runtime expression that specifies the location of the correlation ID.
+
+##### Examples
+
+```json
+{
+  "description": "Default Correlation ID",
+  "type": "string",
+  "location": "$message.header.correlationId"
+}
+```
+
+```yaml
+description: Default Correlation iD
+type: string
+location: $message.header.correlationId
+```
+
+### <a name="runtimeExpression"></a>Runtime Expression
+
+A runtime expression allows values to be defined based on information that will be available within the message.
+This mechanism is used by [Correlation ID Object](#correlationIdObject).
+
+The runtime expression is defined by the following [ABNF](https://tools.ietf.org/html/rfc5234) syntax
+
+```
+      expression = ( "$message" )
+      source = ( header-reference | payload-reference )  
+      header-reference = "header." token
+      payload-reference = "payload." name
+      fragment = a JSON Pointer [RFC 6901](https://tools.ietf.org/html/rfc6901) 
+      name = *( char )
```

Comment:
If `name` allows for "zero" chars, we could end up having something like `$message.payload.`.

What about this?
```suggestion
      name = 1*( char )
```

Reply:
Works for me